src/jit/codegenarm64.cpp

```diff
@@ -3438,42 +3452,45 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
 
     genConsumeOperands(tree);
 
-    emitAttr cmpSize = EA_ATTR(genTypeSize(op1Type));
-
-    assert(genTypeSize(op1Type) == genTypeSize(op2Type));
-
-    if (varTypeIsFloating(op1Type))
+    if ((tree->gtFlags & GTF_USE_FLAGS) == 0)
```

Comment:
This is consistent with use of flags for XARCH.  

However, IMO, this flag is not really necessary and should be inverted and called `GTF_SET_FLAGS`

Reply:
It's really neither, flags are set by the ADD/SUB/AND instruction and used by the conditional branch instruction. This is the result of this solution being a hack :)

Reply:
Yes, but when they are not set by the ADD/SUB/AND instruction, they are set by the compare and used by the conditional branch instruction

So using `GTF_SET_FLAGS` would be consistent when the compare should be setting flags.

Reply:
Hmm? We're talking about the compare instruction, an instruction which is not actually emitted in this case. So it can't set nor use the flags.

Reply:
Which is why it should logically be 
    `if ((tree->gtFlags & GTF_SET_FLAGS) != 0)` (implied by inverted above).  If we are not setting flags it makes no sense to emit a compare instruction.

Reply:
It probably should be `if ((op1-gtFlags & GTF_SET_FLAGS) == 0)`. That is, if op1 doesn't set flags we need to emit a compare instruction.

Reply:
We agreed to leave this consistent with XARCH